Modernization rewrite

pid1 opened the pull request:
This is an attempt to modernize the Netdata Ansible.

This currently supports Debian, Ubuntu, and OpenSUSE, but additional distributions can be supported easily and in a consistent pattern.

Patches welcome.

Fixes https://github.com/netdata/ansible/issues/10
Fixes https://github.com/netdata/ansible/issues/7
Fixes https://github.com/netdata/ansible/issues/12 via deprecation, though we should handle binary installs gracefully.

Deprecates https://github.com/netdata/ansible/pull/8 - I would like to see Molecule tests re-added in the future.

pid1 committed:
Delete upstream’s ansible.cfg and host file

pid1 committed:
Ignore hosts, ansible config. and vscode from Git

pid1 committed:
Make iteration easy

pid1 committed:
Lint fixes

pid1 committed:
Fix naming convention, support aarch64

ralphm commented on .github/CODEOWNERS:
```diff
@@ -1,2 +1,3 @@
-# For the time being, @Ferroin is responsible for everything here.
+# For the time being, @Ferroin is responsible for everything here
```
This is probably an accidental change.

ralphm commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+.vscode
+hosts
+ansible.cfg
+.swp
```
I have an ambivalent approach towards .gitignore files in repos, generally opting for a combination of a well maintained `~/.config/git/ignore` and project specifc `.git/info/exclude` in the working tree. On the other hand, our Agent repo has a very extensive one.

That said, if you want to exclude ViM (swap) files, have a look at https://github.com/github/gitignore/blob/main/Global/Vim.gitignore.

pid1 replied:
I'm equivalently ambivalent about it. This was mainly for my convenience. I'll remove it.

ralphm commented on README.md:
```diff
@@ -1,47 +1,42 @@
-# Example of basic Netdata agent management using Ansible
-## Prerequisites
-Tested with Ansible v. 2.12.1; should work with any Ansible version since 2.9
+# Netdata Ansible
```
At some point, possibly not as part of _this_ PR, but definitely before publishing this to Galaxy, we need to expand this (and also https://learn.netdata.cloud/) with all the current tweakables and expand example usage.

ralphm commented on defaults/main.yml:
```diff
@@ -0,0 +1,43 @@
+---
+role_version: 1.0.0
```
Possibly we should have 0.x.x here. Let's have 1.0.0 when we do an actual release. I also don't think that information goes here, but in `meta/main.yml` in a `galaxy_info` object, or in `galaxy.yml`?

ralphm commented on defaults/main.yml:
```diff
@@ -0,0 +1,43 @@
+role_version: 1.0.0
+
+# Define the Netdata release version we install
+netdata_release_version: "stable"
```
We call this _release channel_.

Also, the names we typically use are `stable` and `nightly`, but the the native build repos use `edge` instead of `nightly`. I think it would be good so be able to put `nightly` as a value here and it doing the right thing.

pid1 replied:
@ralphm I'll update this to reflect the release channel terminology.

As for nightly versus stable, I'd suggest considering the target audience for the Ansible versus the command-line installer. The script defaulting to nightly is unexpected and surprising as a user, and I'd certainly be displeased as a sysadmin if Ansible auto-installed a nightly/non-stable version by default.

Happy to be convinced otherwise here.

ralphm replied:
Oh, I did not mean having nightly as the default. Just that it can take that as a value, instead of, or next to edge.

ralphm commented on tasks/install-debian.yml:
```diff
@@ -0,0 +1,39 @@
+    - name: Install Netdata repo key.
+      ansible.builtin.apt_key:
+        url: "{{ netdata_repository_key_url }}"
+        state: present
```
I think we should explicitly include the key ID here. This ensures that if the key is already present, no retrieval has to be done. I also provides some protection against man-in-the-middle attacks or the URL being compromised.

ralphm commented on tasks/node_claim.yml:
```diff
@@ -0,0 +1,10 @@
+    owner: root
+    group: root
+    mode: '0644'
+  notify: Restart Netdata
```
We don't really need to restart the Agent in order for it to process `claim.conf`. The Netdata CLI has a `reload-claiming-state` command that calls a corresponding API in the Agent to just reload its config and (re-)connect to Cloud.

It might be nice to prevent restarting Netdata if we don't need to because of other `notify` calls.

pid1 replied:
Ah nice, I didn't realize that occurred. I'll remove the restart.

ralphm replied:
But add the reload as a handler.

ralphm commented on templates/claim.conf.j2:
```diff
@@ -0,0 +1,8 @@
+    {% if netdata_proxy %}
+        proxy = {{ netdata_proxy }}
+    {% endif %}
+    insecure = 'no'
```
Given that `no` is the default and we are not making this into a variable, we can do two things:

1. Just remove it from the template.
2. Have some kind of dict for _other_ options, like `insecure`, providing a way to control it.

I'm favoring option 1 right now.

pid1 replied:
I'll remove that.

ralphm commented on tasks/configure-linux.yml:
```diff
@@ -0,0 +1,47 @@
+
+- name: Lay down charts.d.conf
+  ansible.builtin.copy:
+    content: 'enable_all_charts="yes"'
```
This is the default. Why make this explicit, while not offering any other configuration?

pid1 replied:
Ah, is this enabled by default without `charts.d.conf` existing?

ralphm replied:
Yes

ralphm commented on tasks/install-ubuntu.yml:
```diff
@@ -0,0 +1,37 @@
+---
```
This is identical to `install-debian.yml`. Can we consolidate, e.g. by using `ansible_pkg_mgr`?

ralphm commented on vars/main.yml:
```diff
@@ -0,0 +1,2 @@
+---
+# vars file for role
```
I think we can remove this, if empty.

ralphm reviewed (changes requested):
@pid1 Here is my preliminary feedback, as discussed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/netdata/ansible?pullRequest=16) <br/>All committers have signed the CLA.

sashwathn commented:
@agomerz : Can you please sign the CLA?

rawiriblundell commented:
Hi @agomerz: This PR is blocked, can you please sign the CLA?

Alternatively, @pid1: would you be open to another PR with only your commits cherry-picked?

pid1 commented:
I’ll do the latter

pid1 committed:
Modernize Ansible role: rewrite structure and address PR feedback

Remove legacy multi-role structure (roles/, molecule/, claim.yml, purge.yml,
netdata-agent.yml, .ansible-lint, group_vars/, host_vars/).

Implement single-role layout supporting Debian, Ubuntu, and OpenSUSE with
CI workflows for linting and integration testing.

Address all review feedback from netdata/ansible#16:
- Restore trailing period in CODEOWNERS comment
- Remove .gitignore per reviewer agreement
- Rename netdata_release_version to netdata_release_channel with nightly->edge mapping
- Move role version from defaults to meta/main.yml as 0.1.0
- Add explicit GPG key ID to apt_key task
- Replace restart handler with netdatacli reload-claiming-state for claims
- Remove default insecure=no from claim.conf.j2
- Remove redundant enable_all_charts charts.d.conf task
- Consolidate install-ubuntu.yml into install-debian.yml
- Remove empty vars/main.yml

pid1 committed:
Fix meta/main.yml: move version to root level for schema compliance

The ansible-lint schema[meta] rule does not allow 'version' inside
galaxy_info. Move the field to the root level of meta/main.yml to
satisfy the reviewer's request while passing lint validation.

pid1 commented:
@rawiriblundell  I've re-implemented this in https://github.com/netdata/ansible/pull/18 and addressed the PR feedback. Other feedback welcome and enouraged.
